agent/jvm/src/main/java/org/jolokia/jvmagent/JolokiaServer.java

```diff
@@ -339,25 +339,46 @@ private String detectAgentUrl(HttpServer pServer, JolokiaServerConfig pConfig, S
         int port;
         if (serverAddress != null) {
             realAddress = serverAddress.getAddress();
-            if (realAddress.isAnyLocalAddress()) {
-                try {
-                    realAddress = NetworkUtil.getLocalAddress();
-                } catch (IOException e) {
-                    try {
-                        realAddress = InetAddress.getLocalHost();
-                    } catch (UnknownHostException e1) {
-                        // Ok, ok. We take the original one
-                        realAddress = serverAddress.getAddress();
-                    }
-                }
-            }
             port = serverAddress.getPort();
         } else {
             realAddress = pConfig.getAddress();
             port = pConfig.getPort();
```

Comment:
Just wondering what would happen if the address uses a default port (e.g. 80 or 443) implicitly?  Don't we need to consider a case when the value of port returns `-1`?

Reply:
For `InetAddress` created explicitly indeed we'd have to handle it. But here, this particular `InetAddress` is obtained from `com.sun.net.httpserver.HttpServer#getAddress()`, so we're fine.
Thanks for highlighting!